File: java/ReverseString.java
class ReverseString{
    public static void main(String[] args) {

        char[] x = {'h','e', 'l','l','o',};
        reverseString(x);
    }

    static void reverseString(char[] s) {
        
        int leftIndex = 0;
        int rightIndex = s.length - 1;

        while(leftIndex < rightIndex) {
            char temp = s[leftIndex];
            s[leftIndex] = s[rightIndex];
            s[rightIndex] = temp;

            leftIndex++;
            rightIndex--;
        }

        System.out.println(s);
    }
}

File: javascript/reverse-string.js
function reverseStr(arrStr) {
  let leftIndex = 0;
  let rightIndex = arrStr.length - 1;

  while (leftIndex < rightIndex) {
    let temp = arrStr[leftIndex];
    arrStr[leftIndex] = arrStr[rightIndex];
    arrStr[rightIndex] = temp;

    leftIndex++;
    rightIndex--;
  }
  return arrStr;
}

console.log(reverseStr(['h', 'e', 'l', 'l', 'o'])); // ["o","l","l","e","h"]
console.log(reverseStr(['H', 'a', 'n', 'n', 'a', 'h'])); // ["h","a","n","n","a","H"]
